pkgs/desktops/gnome/default.nix

```diff
@@ -1,6 +1,14 @@
-{ config, pkgs, lib }:
-
-lib.makeScope pkgs.newScope (self: with self; {
+{ config, pkgs, lib, splicePackages, newScope
+, pkgsBuildBuild, pkgsBuildHost, pkgsBuildTarget, pkgsHostHost, pkgsTargetTarget }:
+
+lib.makeScopeWithSplicing splicePackages newScope {
+  selfBuildBuild = pkgsBuildBuild.gnome;
+  selfBuildHost = pkgsBuildHost.gnome;
+  selfBuildTarget = pkgsBuildTarget.gnome;
+  selfHostHost = pkgsHostHost.gnome;
+  selfTargetTarget = pkgsTargetTarget.gnome or {};
+} (self: { }) (spliced0: { })
+(self: with self; {
```

Comment:
Not the `with self;` allows similar issues to creep. I would recommend possibly getting rid of that and using `pkgs*.gnome` and `self` explicitly